src/main/java/com/eclipsesource/v8/utils/MemoryManager.java

```diff
@@ -73,7 +73,7 @@ public int getObjectReferenceCount() {
     public void persist(final V8Value object) {
         v8.getLocker().checkThread();
         checkReleased();
-        references.remove(object);
+        memoryManagerReferenceHandler.v8HandleDisposed(object);
```

Comment:
Can you include a test case in the MemoryManagerTest that gets the same V8ObjectTwice (so there are two copies of it) and then one is persisted? After we release the MemoryManager one should still remain. I can help you with this if you need help.